src/components/search/advancedSearch/advancedSearchDialogPopup/AdvancedSearchDialogPopup.js

```diff
@@ -0,0 +1,38 @@
+import { DialogChecklistBase } from "@/components/search/advancedSearch/advancedSearchHelpers/dialogChecklistBase/DialogChecklistBase";
+import StatefulDialog from "@/components/base/statefulDialog/StatefulDialog";
+import { useEffect, useReducer } from "react";
+
+import {
+  initializeMenuItem,
+  reducerForToggleMenuItemsState,
+} from "@/components/search/advancedSearch/advancedSearchHelpers/dropdownReducerFunctions";
+
+export default function AdvancedSearchDialogPopup({
+  menuItems,
+  indexName,
+  updateIndex,
+}) {
+  menuItems = menuItems.map(initializeMenuItem);
+  const [menuItemsState, toggleMenuItemsState] = useReducer(
+    (currentMenuItems, itemUpdate) =>
+      reducerForToggleMenuItemsState({
+        currentMenuItems: currentMenuItems,
+        itemUpdate: itemUpdate,
+      }),
+    menuItems,
+    undefined
+  );
+
+  useEffect(() => {
+    updateIndex(menuItemsState);
+  }, [JSON.stringify(menuItemsState)]);
```

Comment:
Er der behov for at stringify objektet ?

Reply:
Ja, ellers er det kun referencen til `menuItemsState` som dependency arrayet kommer til at tjekke på, og jeg har brug for, at den opdaterer indekset hver gang indholdet i `menuItemsState` opdateres